superbench/analyzer/diagnosis_rule_op.py

```diff
@@ -66,13 +67,13 @@ def variance(data_row, rule, summary_data_row, details, categories):
             data_row (pd.Series): raw data of the metrics
             rule (dict): rule including function, criteria, metrics with their baseline values and categories
             summary_data_row (pd.Series): results of the metrics processed after the function
-            details (list): defective details including data and rules
+            details (list): details about violated rules and related data
             categories (set): categories of violated rules
 
         Returns:
-            bool: whether the rule is passed
+            number: the number of the metrics that violate the rule if the rule is passed, otherwise 0
```

Comment:
what do you mean by `if the rule is passed`? `violate` means `not passed` ?

Reply:
yes